### PR TITLE
State machine shooter

### PR DESCRIPTION
**Pull request by ChrisHirsch**

### CODE REVIEW

File: components/ShooterLogic.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        if not self.SensorArray[0].get():
+            self.ShooterMotors.runLoader(-1*.4)
+            self.next_state_now('runShooterMotor')
+
+    @state
+    def runShooterMotor(self):
+        if self.SensorArray[0].get():
+            self.ShooterMotors.stopLoader()
+            self.ShooterMotors.runShooter(1)
+            if self.ShooterMotors.shooterMotor.getEncoder().getVelocity() >= 5000:
```
Comment (ChrisHirsch) [inline components/ShooterLogic.py]: 5000 should be tunable

Comment (ChrisHirsch) [reply components/ShooterLogic.py]: shooterVelocity = tunable(default=5000)
and then networktables /components/AutomaticShooter/shooterVelocity (I think)

File: robotMap.py [hunk 2]
```diff
@@ -25,4 +24,97 @@ def __init__(self, config):
...
+        self.driveLeftTrig = self.drive.getRawAxis(XboxController.Axis.kLeftTrigger)
+        self.driveAButton = self.drive.getRawButton(XboxController.Button.kA)
+        #Mechanism controller inputs
+        self.mechLeft = self.mech.getRawAxis(XboxController.Axis.kLeftY)
+        self.mechRight = self.mech.getRawAxis(XboxController.Axis.kRightY)
+        self.mechLeftHoriz = self.mech.getRawAxis(XboxController.Axis.kLeftX)
+        self.mechRightHoriz = self.mech.getRawAxis(XboxController.Axis.kRightX)
+        self.mechRightTrig = self.mech.getRawAxis(XboxController.Axis.kRightTrigger)
+        self.mechLeftTrig = self.mech.getRawAxis(XboxController.Axis.kLeftTrigger)
+        self.mechAButton = self.mech.getRawButton(XboxController.Button.kA)
```
Comment (ChrisHirsch) [inline robotMap.py]: Need to remove the buttons here

File: testBot.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from magicbot import MagicRobot
```
Comment (ChrisHirsch) [inline testBot.py]: please move this to tests/ and document what this does and how to use it

File: robotMap.py (hunk 2 above)
Comment (ManiacMoustache) [reply robotMap.py]: These are removed in the other branch, necessary for this branch

File: components/ShooterLogic.py (hunk 1 above)
Comment (ChrisHirsch) [reply components/ShooterLogic.py]: lets do this in another branch shooter-tunable

File: components/driveTrain.py [hunk 4]
```diff
@@ -14,14 +12,7 @@ def setup(self):
...
         self.controllingOverTank = False
-
-        self.motors = {}
-
-        for motorDescKey in self.driveTrain_motorsList:
-            currentMotor = self.driveTrain_motorsList[motorDescKey]
-            print("{}".format(currentMotor))
-            self.motors[motorDescKey] = motorHelper.createMotor(currentMotor)
-
+        self.motors = self.motorsList
```
Comment (tuffinmuffin) [inline components/driveTrain.py]: Did we need to rename this variable? motorsList could be used everwehre instead

Comment (tuffinmuffin) [reply components/driveTrain.py]: Dito for other places with this

File: motorHelper.py [hunk 5]
```diff
@@ -194,19 +204,20 @@ def setupPid(self):
...
-
         self.kPreScale = pid['kPreScale']
         self.PIDController = self.getPIDController() #creates pid controller
 
         self.PIDController.setP(pid['kP'], pid['feedbackDevice'])
         self.PIDController.setI(pid['kI'], pid['feedbackDevice'])
         self.PIDController.setD(pid['kD'], pid['feedbackDevice'])
         self.PIDController.setFF(pid['kF'], pid['feedbackDevice'])
-
+        self.setIdleMode(rev.IdleMode.kBrake)
```
Comment (tuffinmuffin) [inline motorHelper.py]: This should be a configurable parameter.

File: robot.py [hunk 6]
```diff
@@ -1,52 +1,64 @@
-
-"""
-Team 3200 Robot base class
-"""
-from wpilib import XboxController
+# Module imports:
```
Comment (tuffinmuffin) [inline robot.py]: Why did the doc string get delete

File: robot.py [hunk 7]
```diff
@@ -1,52 +1,64 @@
...
+from robotMap import RobotMap, XboxMap
 
 class MyRobot(MagicRobot):
     """
     Base robot class of Magic Bot Type
     """
-
+    shootManual: ManualShooter
+    shootAutomatic: AutomaticShooter
+    ShooterMotors: ShooterMotorCreation
```
Comment (tuffinmuffin) [inline robot.py]: class variable use camelCase

File: robot.py [hunk 8]
```diff
@@ -1,52 +1,64 @@
...
-        self.left = 0
-        self.right = 0
-        self.stick = XboxController(0)
-        
-        self.driveTrain_motorsList = dict(self.map.motorsMap.driveMotors)
-        self.mult = 1 #Multiplier for values. Should not be over 1.
+        self.xboxMap = XboxMap(XboxController(0), XboxController(1))
+        self.motorsList = dict(self.map.motorsMap.driveMotors)
+        # Sensor object
+        self.sensorObjects = dio
```
Comment (tuffinmuffin) [inline robot.py]: Why are we assigning a module to a class variable?

File: robot.py [hunk 9]
```diff
@@ -1,52 +1,64 @@
...
 
     def teleopInit(self):
-        #register button events
-        self.buttonManager.registerButtonEvent(self.stick, XboxController.Button.kA, ButtonEvent.kOnPress, exampleCallback)
-        self.buttonManager.registerButtonEvent(self.stick, XboxController.Button.kBack, ButtonEvent.kOnPress | ButtonEvent.kOnRelease, crashCallback)
-        self.buttonManager.registerButtonEvent(self.stick, XboxController.Button.kStart,  ButtonEvent.kWhilePressed, simpleCallback)
-        self.buttonManager.registerButtonEvent(self.stick, XboxController.Button.kY,  ButtonEvent.kOnPress, actionCallback)
+        self.mult = .5 # Multiplier for drive values. Should not be over 1.
```
Comment (tuffinmuffin) [inline robot.py]: Why is this here? Also name is less than clear, mult..multiply? Multiply What? Not tunable?

File: robotMap.py (hunk 2 above)
Comment (ChrisHirsch) [reply robotMap.py]: Ok doofbuttonmanager has these removed

File: components/driveTrain.py (hunk 4 above)
Comment (ChrisHirsch) [reply components/driveTrain.py]: @DrKittyKat

File: testBot.py (hunk 3 above)
Comment (ChrisHirsch) [reply testBot.py]: moved to tests/electronics/testSensors.py

File: robot.py (hunk 6 above)
Comment (ChrisHirsch) [reply robot.py]: Fixed the missing docs

File: motorHelper.py (hunk 5 above)
Comment (ManiacMoustache) [reply motorHelper.py]: It now is

Comment (ManiacMoustache) [reply motorHelper.py]: via IdleBrake: True/False

File: robot.py (hunk 8 above)
Comment (ChrisHirsch) [reply robot.py]: @cannedlock

File: robot.py (hunk 9 above)
Comment (ChrisHirsch) [reply robot.py]: @cannedlock
